### PR TITLE
solution

### PR DESCRIPTION
**Pull request by andreyysak**
[DEMO LINK](https://andreyysak.github.io/react_todo-app/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,156 +1,166 @@
...
+    e.preventDefault();
+
+    setTodos([
+      ...todos,
+      {
+        id: Date.now(),
+        title: query.trim(),
+        completed: false,
+      },
+    ]);
```
Comment (SanyaBratashchuk) [inline src/App.tsx:36]:
```suggestion
    setTodos(prevTodos => [
      ...prevTodos,
      {
        id: Date.now(),
        title: query.trim(),
        completed: false,
      },
    ]);
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,156 +1,166 @@
...
+    }
+  };
+
+  const editTodo = (id: number, newTitle: string) => {
+    setTodos(
+      todos.map(todo =>
+        todo.id === id
+          ? { ...todo, title: newTitle.trim(), isEditing: false }
+          : todo,
+      ),
```
Comment (SanyaBratashchuk) [inline src/App.tsx:51]:
```suggestion
    setTodos(prevTodos =>
      prevTodos.map(todo =>
        todo.id === id
          ? { ...todo, title: newTitle.trim(), isEditing: false }
          : todo,
      ),
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,156 +1,166 @@
...
+  const filteredTodos = useMemo(() => {
+    let filtered = todos;
+
+    if (filterType === 'Active') {
+      filtered = todos.filter(todo => !todo.completed);
+    }
+
+    if (filterType === 'Completed') {
+      filtered = todos.filter(todo => todo.completed);
+    }
```
Comment (SanyaBratashchuk) [inline src/App.tsx:76]: use switch case here. Also create an enum for todo statuses

File: src/components/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  handleFilterChange,
+  filterType,
+  activeTodosCount,
+  todosCount,
+  clearCompleted,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {activeTodosCount} items left
```
Comment (SanyaBratashchuk) [inline src/components/Footer.tsx:23]: handle plural form it should be `1 item left` but `2 items left`

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Consider fixing the layout. 
Screenshot 2024-04-24 at 15 48 35
Screenshot 2024-04-24 at 15 49 29
Screenshot 2024-04-24 at 15 50 28
The title of the edited todo in the input changes to the previous on the next edit

**Review by etojeDenys — CHANGES_REQUESTED** [review]
1. looks too big
image

2. i can add empty todos, let's fix that
image

3. Did you choose not to use context here?
image

**Pull request closed without merging**